tests/sql/restore_incidents_test.go

```diff
@@ -0,0 +1,254 @@
+package sql_test
+
+import (
+	"context"
+	"crypto/rand"
+	"fmt"
+	"github.com/creasty/defaults"
+	"github.com/icinga/icinga-notifications/internal/config"
+	"github.com/icinga/icinga-notifications/internal/event"
+	"github.com/icinga/icinga-notifications/internal/incident"
+	"github.com/icinga/icinga-notifications/internal/object"
+	baseConfig "github.com/icinga/icingadb/pkg/config"
+	"github.com/icinga/icingadb/pkg/icingadb"
+	"github.com/icinga/icingadb/pkg/logging"
+	"github.com/icinga/icingadb/pkg/types"
+	"github.com/pkg/errors"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"go.uber.org/zap"
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+)
+
+func TestIncidentLoading(t *testing.T) {
+	t.Parallel()
+
+	c, err := getDBConfig()
+	if err != nil {
+		t.Skipf("Skipping test due to an error: %s", err.Error())
+	}
+
+	db, err := c.Database.Open(logging.NewLogger(zap.NewNop().Sugar(), time.Hour))
+	require.NoError(t, err, "connecting to database should not fail")
+
+	ctx := context.Background()
+	if c.Schema != "" { // Import schema when requested to do it!
+		_, err = db.ExecContext(ctx, c.Schema)
+		require.NoError(t, err, "schema import should not fail")
+	}
```

Comment:
I think I'd prefer importing the schema on the outside, i.e. in the GHA workflow definition. With PostgreSQL this is probably fine, but with MySQL, this will likely become annoying (see the workarounds we do in `icinga-testing` as the `mysql` CLI client parses a schema file differently than the MySQL server parses statements sent to it).

Reply:
However, when using Github Actions for this, I have to somehow perform `\c DATABASE` command before importing the schema, something I have not yet managed to do, so I've added this fallback.

Reply:
Shouldn't it just do the right thing it you set `POSTGRES_DB` and put the schema into something like `/docker-entrypoint-initdb.d/schema.sql`?

Reply:
No, it doesn't seem to be the case!
https://github.com/Icinga/icinga-notifications/actions/runs/8734094424/job/23964046786#step:2:143

Reply:
> psql:/docker-entrypoint-initdb.d/notifications.sql: error: could not read from input file: Is a directory

Looks like something went wrong with mounting the schema file into the container?

Reply:
Didn't get that log to see, but yeah, I wasn't aware that `github.workspace` is just an empty directory until the `checkout` step gets executed.